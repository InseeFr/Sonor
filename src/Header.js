import React from 'react';
import logo from './logo.jpeg';



class Header extends React.Component {

  render() {
    return (
      <header id='App-header'>
        <img src={logo} id='InseeLogo' alt='logo' />
        <button className='HeaderButton RedButton'>Affecter</button>
        <button className='HeaderButton RedButton'>Suivre</button>
        <button className='HeaderButton RedButton'>Contrôler</button>
        <UserZone userName={this.props.keycloak.idTokenParsed.name} date='08/06/2020'/>
      </header>
    );
  }
}

class UserZone extends React.Component {

  render() {
    return (
      <div id='UserZone'>
        <div>Bienvenue {this.props.userName}</div>
        <div>{this.props.date}</div>
        <div className='UserZoneButtons'>
          <button className='HeaderButton YellowButton'>Mes enquêteurs</button>
          <button className='HeaderButton YellowButton'>Mes enquêtes</button>
        </div>
      </div>
    );
  }
}


export default Header;
