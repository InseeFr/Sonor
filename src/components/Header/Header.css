.selectedSubeMenu:hover {
    color: #fff;
    background-color: #0069d9;
    border-color: #0062cc;
    text-decoration: none !important;
    margin: 0 !important;

}
a {
    display: block;
    padding: 0 16px;
    line-height: inherit;
    cursor: pointer;
    }

ul {
    list-style: none;
    padding-left: 0;
    margin-top: 0;
    margin-bottom: 0;
    }

li {
    list-style: none;
    }

 .dropdown-menu .sub-menu {
    left: 100%;
    position: absolute;
    top: 0;
    margin-top: -1px;
  }
  
  .dropdown-menu{
    display: block !important;
  }

  .nav-tabs .dropdown-menu,
  .nav-pills .dropdown-menu,
  .navbar .dropdown-menu {
    margin-top: 0;
  }
  
  .navbar .sub-menu:before {
    border-bottom: 7px solid transparent;
    border-left: none;
    border-right: 7px solid rgba(0, 0, 0, 0.2);
    border-top: 7px solid transparent;
    left: -7px;
    top: 10px;
  }
  
  .navbar .sub-menu:after {
    border-top: 6px solid transparent;
    border-left: none;
    border-right: 6px solid #fff;
    border-bottom: 6px solid transparent;
    left: 10px;
    top: 11px;
    left: -6px;
  } 