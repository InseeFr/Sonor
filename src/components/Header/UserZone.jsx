import React from 'react';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import D from '../../i18n';

function UserZone({ user, date, showPreferences }) {
  return (
    <Card id="UserZone">
      <Card.Title>
        { 'Bienvenue '}
        {user.firstName}
        &nbsp;
        {user.lastName}
      </Card.Title>
      <Card.Subtitle className="mb-2 text-muted">{date.toLocaleDateString()}</Card.Subtitle>
      <div className="UserZoneButtons">
        <Row>
          <Col xs="8">
            <Button className="HeaderButton" onClick={() => showPreferences()}>{D.mySurveys}</Button>
          </Col>
          <Col xs="4">
            <div
              className="HeaderDocLink Clickable"
              onClick={() => window.open('', '_blank')}
              role="link"
              tabIndex="0"
            >
              <i className="fa fa-question-circle fa-2x" />
            </div>
          </Col>
        </Row>
      </div>
    </Card>
  );
}

export default UserZone;
