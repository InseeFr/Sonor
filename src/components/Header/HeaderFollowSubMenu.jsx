import React, { useEffect, useRef } from 'react';
import { Link } from 'react-router-dom';
import D from '../../i18n';
import './Header.css';

function HeaderFollowSubMenu({ toggle, setModal }) {
  function useOutsideAlerter(ref) {
    useEffect(() => {
      function handleClickOutside(event) {
        if (ref.current && !ref.current.contains(event.target)) {
          toggle(false);
        }
      }
      document.addEventListener('mousedown', handleClickOutside);
      return () => {
        document.removeEventListener('mousedown', handleClickOutside);
      };
    }, [ref]);
  }

  const elmRef = useRef(null);
  useOutsideAlerter(elmRef);

  return (
    <ul ref={elmRef} className="dropdown-menu">
      <li>
        <span className="selectedSubeMenu">
          {D.site}
          <i className="fa fa-caret-right fa-xs subMenucaret" />
        </span>
        <ul className="dropdown-menu sub-menu">
          <li>
            <Link
              to="/follow/campaigns"
              className="selectedSubeMenu"
              data-testid="follow-by-survey"
              onClick={() => toggle(false)}
            >
              {D.progression}
            </Link>
          </li>
          <li>
            <Link
              to="/collection/campaigns"
              className="selectedSubeMenu"
              data-testid="collection-by-survey"
              onClick={() => toggle(false)}
            >
              {D.collection}
            </Link>
          </li>
        </ul>
      </li>
      <li>
        <span
          className="selectedSubeMenu"
          data-testid="survey-sub-menu-line"
        >
          {D.survey}
          <i className="fa fa-caret-right fa-xs subMenucaret" />
        </span>
        <ul className="dropdown-menu sub-menu secondDropdownRow">
          <li>
            <span className="selectedSubeMenu">
              {D.table}
              <i className="fa fa-caret-right fa-xs subMenucaret" />
            </span>
            <ul className="dropdown-menu sub-menu-level-2 larger-sub-menu">
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="progress-by-site"
                  onClick={() => setModal(D.progressBySite, 'follow/sites')}
                  tabIndex={0}
                >
                  {D.progressBySite}
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="follow-by-interviewer"
                  onClick={() => setModal(D.progressByInterviewer, 'follow/campaign')}
                  tabIndex={0}
                >
                  {D.progressByInterviewer}
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="collection-by-site"
                  onClick={() => setModal(D.collectionBySite, 'collection/sites')}
                  tabIndex={0}
                >
                  {D.collectionBySite}
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="collection-by-interviewer"
                  onClick={() => setModal(D.collectionByInterviewer, 'collection/campaign')}
                  tabIndex={0}
                >
                  {D.collectionByInterviewer}
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="provisional-status-by-interviewer"
                  onClick={() => setModal(D.provisionalStatusByInterviewer, 'provisionalstatus/campaign')}
                  tabIndex={0}
                >
                  {D.provisionalStatusByInterviewer}
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  onClick={() => setModal(D.unitsDone, 'terminated')}
                  tabIndex={0}
                >
                  {D.unitsDone}
                </button>
              </li>
            </ul>
          </li>
          <li>
            <span className="selectedSubeMenu">
              {D.graph}
              <i className="fa fa-caret-right fa-xs subMenucaret" />
            </span>
            <ul className="dropdown-menu sub-menu-level-2 larger-sub-menu secondDropdownRow">
              <li>
                <Link
                  to="/"
                  className="selectedSubeMenu"
                  onClick={() => toggle(false)}
                >
                  {D.siteProgress}
                </Link>
              </li>
              <li>
                <Link
                  to="/"
                  className="selectedSubeMenu"
                  onClick={() => toggle(false)}
                >
                  {D.siteCollection}
                </Link>
              </li>
            </ul>
          </li>
        </ul>
      </li>
      <li>
        <span id="selectedSubeMenu" className="selectedSubeMenu">
          {D.interviewer}
          <i className="fa fa-caret-right fa-xs subMenucaret" />
        </span>
        <ul className="dropdown-menu sub-menu thirdDropdownRow">
          <li>
            <span className="selectedSubeMenu">
              {D.table}
              <i className="fa fa-caret-right fa-xs subMenucaret" />
            </span>
            <ul className="dropdown-menu sub-menu-level-2 larger-sub-menu">
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="progress-by-survey-one-interviewer"
                  onClick={() => setModal(D.progressBySurvey, 'follow/campaigns/interviewer', true)}
                  tabIndex={0}
                >
                  {D.progressBySurvey}
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="collection-by-survey-one-interviewer"
                  onClick={() => setModal(D.collectionBySurvey, 'collection/campaigns/interviewer', true)}
                  tabIndex={0}
                >
                  {D.collectionBySurvey}
                </button>
              </li>
              <li>
                <button
                  type="button"
                  className="selectedSubeMenu"
                  data-testid="provisional-status-by-survey-one-interviewer"
                  onClick={() => setModal(D.provisionalStatusBySurvey, 'provisionalstatus/campaigns/interviewer', true)}
                  tabIndex={0}
                >
                  {D.provisionalStatusBySurvey}
                </button>
              </li>
            </ul>
          </li>
          <li>
            <span className="selectedSubeMenu">
              {D.graph}
              <i className="fa fa-caret-right fa-xs subMenucaret" />
            </span>
            <ul className="dropdown-menu sub-menu-level-2 secondDropdownRow">
              <li>
                <Link
                  to="/"
                  className="selectedSubeMenu"
                  onClick={() => toggle(false)}
                >
                  {D.progression}
                </Link>
              </li>
              <li>
                <Link
                  to="/"
                  className="selectedSubeMenu"
                  onClick={() => toggle(false)}
                >
                  {D.collection}
                </Link>
              </li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  );
}

export default HeaderFollowSubMenu;
