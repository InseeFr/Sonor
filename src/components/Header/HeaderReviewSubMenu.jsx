import React, { useEffect, useRef } from 'react';
import { Link } from 'react-router-dom';
import D from '../../i18n';
import './Header.css';

function HeaderReviewSubMenu({ toggle, setModal }) {
  function useOutsideAlerter(ref) {
    useEffect(() => {
      function handleClickOutside(event) {
        if (ref.current && !ref.current.contains(event.target)) {
          toggle(false);
        }
      }
      document.addEventListener('mousedown', handleClickOutside);
      return () => {
        document.removeEventListener('mousedown', handleClickOutside);
      };
    }, [ref]);
  }

  const elmRef = useRef(null);
  useOutsideAlerter(elmRef);

  return (
    <ul ref={elmRef} className="dropdown-menu larger-sub-menu">
      <li>
        <Link
          to="/review"
          className="selectedSubeMenu"
          data-testid="review-link"
          onClick={() => toggle(false)}
        >
          {D.unitsOfSite}
        </Link>
      </li>
      <li>
        <button
          type="button"
          className="selectedSubeMenu"
          onClick={() => setModal(D.unitsOfCampaign, 'review')}
          tabIndex={0}
        >
          {D.unitsOfCampaign}
        </button>
      </li>
    </ul>
  );
}

export default HeaderReviewSubMenu;
