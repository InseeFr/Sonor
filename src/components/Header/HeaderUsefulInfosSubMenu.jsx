import React, { useEffect, useRef } from 'react';
import D from '../../i18n';
import './Header.css';

function HeaderUsefulInfosSubMenu({ toggle, setModal }) {
  function useOutsideAlerter(ref) {
    useEffect(() => {
      function handleClickOutside(event) {
        if (ref.current && !ref.current.contains(event.target)) {
          toggle(false);
        }
      }
      document.addEventListener('mousedown', handleClickOutside);
      return () => {
        document.removeEventListener('mousedown', handleClickOutside);
      };
    }, [ref]);
  }

  const elmRef = useRef(null);
  useOutsideAlerter(elmRef);

  return (
    <ul ref={elmRef} className="dropdown-menu larger-sub-menu">
      <li>
        <button
          type="button"
          className="selectedSubeMenu"
          onClick={() => setModal(D.collectionOrganization, 'portal')}
          tabIndex={0}
        >
          {D.collectionOrganization}
        </button>
      </li>
      <li>
        <button
          type="button"
          className="selectedSubeMenu"
          onClick={() => setModal(D.unitsAllocatedToSite, 'listSU')}
          tabIndex={0}
        >
          {D.unitsAllocatedToSite}
        </button>
      </li>
    </ul>
  );
}

export default HeaderUsefulInfosSubMenu;
