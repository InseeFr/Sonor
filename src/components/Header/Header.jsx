import React, { useState } from 'react';
import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import { Link, useLocation } from 'react-router-dom';
import logo from './logo_com_externe_semi_bold.png';
import UserZone from './UserZone';
import D from '../../i18n';
import { version } from '../../../package.json';

import './Header.css';

function Header({
  user, showPreferences,
}) {
  const [showDropdown, setShowDropdown] = useState(false);
  const { pathname } = useLocation();

  return (
    <header id="App-header" className="shadow">
      <Container fluid>
        <Row>
          <Col md="auto">
            <Link to="/" className="ButtonLink">
              <img
                src={logo}
                id="InseeLogo"
                alt="logo"
              />
              <div id="appVersion">
                {version}
              </div>
            </Link>
          </Col>
          <Col>
            <div className="d-inline-flex classTest" id="headerButtonContainer">
              <Link to="/followUp" className="ButtonLink">
                <Button
                  className={`HeaderButton HeaderFocusableButton ${pathname.includes('/followUp') ? ' ButtonActive' : ''}`}
                >
                  {D.remind}
                </Button>
              </Link>
              <li className="dropdown" id="BtnSuivreParent">
                <Button
                  id="FollowButton"
                  className={`HeaderButton HeaderFocusableButton dropdown-toggle ${pathname.includes('/follow/') ? ' ButtonActive' : ''}`}
                  data-toggle="dropdown"
                  onClick={() => { setShowDropdown(!showDropdown); }}
                >
                  {D.follow}
                  <b className="caret" />
                </Button>
                {!showDropdown || displayFirstSubMenu(setShowDropdown)}
              </li>
              <Link to="/review" className="ButtonLink">
                <Button
                  className={`HeaderButton HeaderFocusableButton ${pathname.includes('/review') ? ' ButtonActive' : ''}`}
                >
                  {D.read}
                </Button>
              </Link>
            </div>
          </Col>
          <Col><UserZone user={user} date={new Date()} showPreferences={showPreferences} /></Col>
        </Row>
      </Container>
    </header>
  );
}

function displayFirstSubMenu(toggle) {
  return (
    <ul className="dropdown-menu" onClick={() => toggle(false)}>
      <li>
        <span className="selectedSubeMenu">
          {D.surveys}
          <i className="fa fa-caret-right fa-xs subMenucaret" />
        </span>
        <ul className="dropdown-menu sub-menu">
          <li>
            <Link to="/follow/campaigns" className="selectedSubeMenu">
              {D.progression}
            </Link>
          </li>
        </ul>
      </li>
      <li>
        <span id="selectedSubeMenu" className="selectedSubeMenu">
          {D.interviewers}
          <i className="fa fa-caret-right fa-xs subMenucaret" />
        </span>
        <ul className="dropdown-menu sub-menu" id="SecondDropdown">
          <li>
            <Link to="/follow/interviewers" className="selectedSubeMenu">
              {D.progression}
            </Link>
          </li>
        </ul>
      </li>
    </ul>
  );
}

export default Header;
