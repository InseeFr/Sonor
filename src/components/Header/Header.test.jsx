// Link.react.test.js
import React from 'react';
import {
  render, screen, cleanup, fireEvent, waitForElement,
} from '@testing-library/react';
import {
  Router, Route, Switch,
} from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Header from './Header';
import mocks from '../../tests/mocks';

import DataFormatter from '../../utils/DataFormatter';
import C from '../../utils/constants.json';

jest.mock(
	"../../../package.json",
	() => ({
		version: "1.0.0",
	})
);


const {
  mainScreenData,
  userData,
  preferences,
  respModeByInterviewers1Survey,
  respModeBySite,
  respModeBySurvey,
  respModeByInterviewer,
  reviewDataAllSurveys,
  campaignPortalData,
  suTerminated,
  listSU,
  interviewers,
} = mocks;

const history = createMemoryHistory();

const toLocaleDateString = Date.prototype.toLocaleString;
Date.prototype.toLocaleDateString = function() {
  return toLocaleDateString.call(this, 'en-EN', { timeZone: 'UTC',year: "numeric", month: "numeric", day: "numeric" });
};
const OriginalDate = global.Date;
jest
  .spyOn(global, 'Date')
  .mockImplementation((a) => (a ? new OriginalDate(a) : new OriginalDate('2020-08-20T11:01:58.135Z')));
Date.now = jest.fn(() => 1597916474000);

jest.mock('../../utils/DataFormatter');
DataFormatter.mockImplementation(() => ({
  getPreferences: (c) => (c(preferences)),
  getInterviewers: (c) => (c(interviewers)),
  getQuestionnaireId: (id, c) => (c({ questionnaireId: 'QXT55' })),
  getFormattedCampaignsForMainScreen: (a, c, mainScreenData) => {
    if (c) { c(mainScreenData); }
    return Promise.resolve(mainScreenData);
  },
  getListSuTerminated: (id, cb) => (cb(suTerminated)),
  getDataForReview: (s, cb) => (cb(reviewDataAllSurveys)),
  getDataForCampaignPortal: (a, cb) => (cb(campaignPortalData)),
  getDataForListSU: (a, cb) => (cb(listSU)),
  getUserInfo: (cb) => (cb(userData)),
  getDataForMonitoringTable: (survey, date, pagination, mode, cb) => {
    switch (mode) {
      case C.BY_INTERVIEWER_ONE_SURVEY:
        cb(respModeByInterviewers1Survey);
        break;
      case C.BY_SITE:
        cb(respModeBySite);
        break;
      case C.BY_SURVEY:
        cb(respModeBySurvey);
        break;
      case C.BY_INTERVIEWER:
        cb(respModeByInterviewer);
        break;
      default:
        break;
    }
  },
}));

const dataRetreiver = new DataFormatter();

beforeEach(() => {
  history.push('/');
});

afterEach(cleanup);

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => jest.requireActual('react-router-dom').useLocation().location || jest.requireActual('react-router-dom').useLocation(),
}));

const TestingRouter = () => (
  <Router history={history}>
    <Header user={userData} showPreferences={() => { mockShowPreferences(); }} dataRetreiver={dataRetreiver} campaigns={mainScreenData} preferences={preferences}/>
    <Switch>
      <Route
        path="*"
        render={(routeProps) => (
          <div>
            <div data-testid="Redirect-url">{JSON.stringify(routeProps.history.location.pathname)}</div>
          </div>
        )}
      />
    </Switch>
  </Router>
);

const mockShowPreferences = jest.fn();

it('Component is correctly displayed', async () => {
  const component = render(
    <TestingRouter />,
  );
  // Should match snapshot (name is ChloÃ© Dupont, date is displayed etc...)
  expect(component).toMatchSnapshot();
});

it('Go to monitoring table by survey', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('follow-by-survey').click();

  // Should redirect to /follow/campaigns
  const redirectUrl = '/follow/campaigns';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to monitoring table by interviewer', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('follow-by-interviewer').click();

  fireEvent.change(component.getByTestId('Survey_selector'), { target: { value: 'simpsons2020x00' } });

  // Should redirect to /follow/interviewers
  const redirectUrl = '/follow/campaign/simpsons2020x00';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to monitoring table by site', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('progress-by-site').click();

  fireEvent.change(component.getByTestId('Survey_selector'), { target: { value: 'simpsons2020x00' } });

  // Should redirect to /follow/interviewers
  const redirectUrl = '/follow/sites/simpsons2020x00';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to monitoring table by survey one interviewer', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('progress-by-survey-one-interviewer').click();

  await waitForElement(() => screen.getByTestId('Interviewer_selector'));

  fireEvent.change(component.getByTestId('Interviewer_selector'), { target: { value: 'INTW5' } });

  // Should redirect to /follow/interviewers
  const redirectUrl = '/follow/campaigns/interviewer/INTW5';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to collection table by survey', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('collection-by-survey').click();

  // Should redirect to /follow/campaigns
  const redirectUrl = '/collection/campaigns';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to collection table by interviewer', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('collection-by-interviewer').click();

  fireEvent.change(component.getByTestId('Survey_selector'), { target: { value: 'simpsons2020x00' } });

  // Should redirect to /follow/interviewers
  const redirectUrl = '/collection/campaign/simpsons2020x00';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to collection table by site', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('collection-by-site').click();

  fireEvent.change(component.getByTestId('Survey_selector'), { target: { value: 'simpsons2020x00' } });

  // Should redirect to /follow/interviewers
  const redirectUrl = '/collection/sites/simpsons2020x00';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to collection table by survey one interviewer', async () => {
  const component = render(
    <TestingRouter />,
  );
  component.baseElement.querySelector('#FollowButton').click();
  screen.getByTestId('collection-by-survey-one-interviewer').click();

  await waitForElement(() => screen.getByTestId('Interviewer_selector'));


  fireEvent.change(component.getByTestId('Interviewer_selector'), { target: { value: 'INTW5' } });

  // Should redirect to /follow/interviewers
  const redirectUrl = '/collection/campaigns/interviewer/INTW5';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});


it('Go to follow-up', async () => {
  render(
    <TestingRouter />,
  );
  screen.getByTestId('follow-up').click();

  // Should redirect to /followUp
  const redirectUrl = '/followUp';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to review', async () => {
  render(
    <TestingRouter />,
  );
  screen.getByTestId('review').click();
  screen.getByTestId('review-link').click();

  // Should redirect to /review
  const redirectUrl = '/review';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to campaign portal', async () => {
  const component = render(
    <TestingRouter />,
  );
  screen.getByTestId('useful-infos-button').click();
  screen.getByTestId('campaign-portal-link').click();

  fireEvent.change(component.getByTestId('Survey_selector'), { target: { value: 'simpsons2020x00' } });

  // Should redirect to /portal/simpsons2020x00
  const redirectUrl = '/portal/simpsons2020x00';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Go to list su', async () => {
  const component = render(
    <TestingRouter />,
  );
  screen.getByTestId('useful-infos-button').click();
  screen.getByTestId('list-su-link').click();

  fireEvent.change(component.getByTestId('Survey_selector'), { target: { value: 'simpsons2020x00' } });

  // Should redirect to /portal/simpsons2020x00
  const redirectUrl = '/listSU/simpsons2020x00';
  expect(screen.getByTestId('Redirect-url').innerHTML).toEqual(`\"${redirectUrl}\"`);
});

it('Show preferences', async () => {
  render(
    <TestingRouter />,
  );
  screen.getByTestId('preferences').click();

  // showPreferences should be called
  expect(mockShowPreferences).toHaveBeenCalled();
});

it('Click on doc link', async () => {
  const component = render(
    <TestingRouter />,
  );
  window.open = jest.fn();

  component.baseElement.querySelector('.HeaderDocLink').click();

  // window.open should have been called
  expect(window.open).toHaveBeenCalled();

});
