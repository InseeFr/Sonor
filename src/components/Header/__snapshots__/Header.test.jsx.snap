// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is correctly displayed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="shadow"
        id="App-header"
      >
        <div
          class="container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="col-md-auto"
            >
              <a
                class="ButtonLink"
                href="/"
              >
                <img
                  alt="logo"
                  id="InseeLogo"
                  src="logo_com_externe_semi_bold.png"
                />
                <div
                  id="appVersion"
                >
                  0.1.3
                </div>
              </a>
            </div>
            <div
              class="col"
            >
              <div
                class="d-inline-flex classTest"
                id="headerButtonContainer"
              >
                <a
                  class="ButtonLink"
                  href="/followUp"
                >
                  <button
                    class="HeaderButton HeaderFocusableButton  btn btn-primary"
                    data-testid="follow-up"
                    type="button"
                  >
                    Follow up
                  </button>
                </a>
                <li
                  class="dropdown"
                  id="BtnSuivreParent"
                >
                  <button
                    class="HeaderButton HeaderFocusableButton dropdown-toggle  btn btn-primary"
                    data-toggle="dropdown"
                    id="FollowButton"
                    type="button"
                  >
                    Follow
                    <b
                      class="caret"
                    />
                  </button>
                </li>
                <a
                  class="ButtonLink"
                  href="/review"
                >
                  <button
                    class="HeaderButton HeaderFocusableButton  btn btn-primary"
                    data-testid="review"
                    type="button"
                  >
                    Review
                  </button>
                </a>
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="card"
                id="UserZone"
              >
                <div
                  class="card-title h5"
                >
                  Bienvenue 
                  Chloé
                   
                  Dupont
                </div>
                <div
                  class="mb-2 text-muted card-subtitle h6"
                >
                  8/20/2020
                </div>
                <div
                  class="UserZoneButtons"
                >
                  <div
                    class="row"
                  >
                    <div
                      class="col-8"
                    >
                      <button
                        class="HeaderButton btn btn-primary"
                        data-testid="preferences"
                        type="button"
                      >
                        My surveys
                      </button>
                    </div>
                    <div
                      class="col-4"
                    >
                      <div
                        class="HeaderDocLink Clickable"
                        role="link"
                        tabindex="0"
                      >
                        <i
                          class="fa fa-question-circle fa-2x"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </header>
      <div>
        <div
          data-testid="Redirect-url"
        >
          "/"
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <header
      class="shadow"
      id="App-header"
    >
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col-md-auto"
          >
            <a
              class="ButtonLink"
              href="/"
            >
              <img
                alt="logo"
                id="InseeLogo"
                src="logo_com_externe_semi_bold.png"
              />
              <div
                id="appVersion"
              >
                0.1.3
              </div>
            </a>
          </div>
          <div
            class="col"
          >
            <div
              class="d-inline-flex classTest"
              id="headerButtonContainer"
            >
              <a
                class="ButtonLink"
                href="/followUp"
              >
                <button
                  class="HeaderButton HeaderFocusableButton  btn btn-primary"
                  data-testid="follow-up"
                  type="button"
                >
                  Follow up
                </button>
              </a>
              <li
                class="dropdown"
                id="BtnSuivreParent"
              >
                <button
                  class="HeaderButton HeaderFocusableButton dropdown-toggle  btn btn-primary"
                  data-toggle="dropdown"
                  id="FollowButton"
                  type="button"
                >
                  Follow
                  <b
                    class="caret"
                  />
                </button>
              </li>
              <a
                class="ButtonLink"
                href="/review"
              >
                <button
                  class="HeaderButton HeaderFocusableButton  btn btn-primary"
                  data-testid="review"
                  type="button"
                >
                  Review
                </button>
              </a>
            </div>
          </div>
          <div
            class="col"
          >
            <div
              class="card"
              id="UserZone"
            >
              <div
                class="card-title h5"
              >
                Bienvenue 
                Chloé
                 
                Dupont
              </div>
              <div
                class="mb-2 text-muted card-subtitle h6"
              >
                8/20/2020
              </div>
              <div
                class="UserZoneButtons"
              >
                <div
                  class="row"
                >
                  <div
                    class="col-8"
                  >
                    <button
                      class="HeaderButton btn btn-primary"
                      data-testid="preferences"
                      type="button"
                    >
                      My surveys
                    </button>
                  </div>
                  <div
                    class="col-4"
                  >
                    <div
                      class="HeaderDocLink Clickable"
                      role="link"
                      tabindex="0"
                    >
                      <i
                        class="fa fa-question-circle fa-2x"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </header>
    <div>
      <div
        data-testid="Redirect-url"
      >
        "/"
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
